python/ray/data/tests/test_hudi.py

```diff
@@ -0,0 +1,49 @@
+import os
+import zipfile
+
+import pyarrow as pa
+import pytest
+from pytest_lazyfixture import lazy_fixture
+
+import ray
+from ray.data.datasource.path_util import _unwrap_protocol
+
+PYARROW_LE_8_0_0 = tuple(int(s) for s in pa.__version__.split(".") if s.isnumeric()) < (
+    8,
+    0,
+    0,
+)
+pytestmark = pytest.mark.skipif(
+    PYARROW_LE_8_0_0, reason="hudi only supported if pyarrow >= 8.0.0"
```

Comment:
What happens currently if a user tries to use pyarrow <8.0.0? Should we warn / error in the `read_hudi_table` or `HudiDatasource` if the user's pyarrow version is less than 8.0.0?

Reply:
IMO, we should raise an exception when using either read_hudi_table or HudiDatasource if the arrow version is not supported

Reply:
thanks for pointing out. i'll add a check and raise exception from within `HudiTable` API since this is internal to Hudi's implementation

Reply:
I think that sounds reasonable too. Just to make sure I understand the idea would be that in the creation of the `HudiDatasource` we would then get an exception from the `from hudi import HudiTable` if the version of pyarrow is unsupported?

Reply:
@omatthew98 that's correct.

Reply:
we have enforced a version requirement here 
https://github.com/apache/hudi-rs/blob/4766375ec91958a593410e8327346d37ca2e7ff4/python/pyproject.toml#L37

I've updated this test module-level checking to using 11.0.0